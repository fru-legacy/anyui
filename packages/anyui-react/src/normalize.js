import * as XmlReader from 'xml-reader';

const reader = XmlReader.create();

export function toJSON(xml) {
    var result = XmlReader.parseSync(xml, {parentNodes: false});
    return strip(result);
}

function strip(result) {
    if (result) {
        if (result.attributes && Object.keys(result.attributes).length === 0) {
            delete result.attributes;
        }
    }

    if (result.children) {
        if (result.children.length === 0) {
            delete result.children;
        } else {
            result.children = result.children.map(strip);
        }
    }

    delete result.parent;
    if (!result.value) delete result.value;
    if (!result.name) delete result.name;
    
    // call recursivly
    return result;
}