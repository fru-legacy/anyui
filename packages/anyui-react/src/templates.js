export function renderJson(React, json) {
    return () => <div>Test</div>;
}