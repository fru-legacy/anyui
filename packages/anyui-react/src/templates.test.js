import {toJSON} from './normalize';
import {renderJson} from './templates';
import React from 'react';
import renderer from 'react-test-renderer';

import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('render', () => {
	it('Should respect the if attribute', () => {
        var dom = renderJson(React, toJSON(`<div if="test > 2">Test</div>`));
        expect(renderer.create(dom({test: 1})).toJSON()).toMatchSnapshot();
        expect(renderer.create(dom({test: 3})).toJSON()).toMatchSnapshot();
        expect(shallow(dom({test: 3})).text()).toBe('Test');
    });
}); 