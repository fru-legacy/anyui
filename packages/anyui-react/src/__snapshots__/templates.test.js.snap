// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Should respect the if attribute 1`] = `
<div>
  Test
</div>
`;

exports[`render Should respect the if attribute 2`] = `
<div>
  Test
</div>
`;
